Parsers/ASimFileEvent/Parsers/vimFileEventSentinelOne.yaml

```diff
@@ -0,0 +1,161 @@
+  let EventTypeLookup = datatable (alertInfo_eventType_s: string, EventType: string)
+      [
+      "FILECREATION", "FileCreated",
+      "FILEMODIFICATION", "FileModified",
+      "FILEDELETION", "FileDeleted",
+      "FILERENAME", "FileRenamed"
+  ];
+  let ThreatConfidenceLookup_undefined = datatable(
+      alertInfo_analystVerdict_s: string,
+      ThreatConfidence_undefined: int
+  )
+      [
+      "FALSE_POSITIVE", 5,
+      "Undefined", 15,
+      "SUSPICIOUS", 25,
+      "TRUE_POSITIVE", 33 
+  ];
+  let ThreatConfidenceLookup_suspicious = datatable(
+      alertInfo_analystVerdict_s: string,
+      ThreatConfidence_suspicious: int
+  )
+      [
+      "FALSE_POSITIVE", 40,
+      "Undefined", 50,
+      "SUSPICIOUS", 60,
+      "TRUE_POSITIVE", 67 
+  ];
+  let ThreatConfidenceLookup_malicious = datatable(
+      alertInfo_analystVerdict_s: string,
+      ThreatConfidence_malicious: int
+  )
+      [
+      "FALSE_POSITIVE", 75,
+      "Undefined", 80,
+      "SUSPICIOUS", 90,
+      "TRUE_POSITIVE", 100 
+  ];
+  let parser = (disabled: bool=false) {
+      let allFileData = SentinelOne_CL
+          | where not(disabled)
+              and event_name_s == "Alerts."
+              and alertInfo_eventType_s in ('FILECREATION', 'FILEMODIFICATION', 'FILEDELETION', 'FILERENAME');
+      let windowsFileData = allFileData
+          | where agentDetectionInfo_osFamily_s == "windows"
+          | extend
+              TargetFilePathType = "Windows Local",
+              TargetFileName = GetWindowsFilenamePart(targetProcessInfo_tgtFilePath_s),
+              SrcFileName = GetWindowsFilenamePart(targetProcessInfo_tgtFileOldPath_s);
+      let otherFileData = allFileData
+          | where agentDetectionInfo_osFamily_s != "windows"
+          | extend
+              TargetFilePathType = "Unix",
+              TargetFileName = GetLinuxFilenamePart(targetProcessInfo_tgtFilePath_s),
+              SrcFileName = GetLinuxFilenamePart(targetProcessInfo_tgtFileOldPath_s);
+      let parseddata = union windowsFileData, otherFileData
+          | lookup EventTypeLookup on alertInfo_eventType_s;
+      let undefineddata = parseddata
+          | where ruleInfo_treatAsThreat_s == "UNDEFINED"
+          | lookup ThreatConfidenceLookup_undefined on alertInfo_analystVerdict_s;
+      let suspiciousdata = parseddata
+          | where ruleInfo_treatAsThreat_s == "Suspicious"
+          | lookup ThreatConfidenceLookup_suspicious on alertInfo_analystVerdict_s;
+      let maaliciousdata = parseddata
+          | where ruleInfo_treatAsThreat_s == "Malicious"
+          | lookup ThreatConfidenceLookup_malicious on alertInfo_analystVerdict_s;
+      union undefineddata, suspiciousdata, maaliciousdata
+      | extend
+          ThreatConfidence = coalesce(ThreatConfidence_undefined, ThreatConfidence_suspicious, ThreatConfidence_malicious),
+          EventSeverity = iff(ruleInfo_severity_s == "Critical", "High", ruleInfo_severity_s),
```

Comment:
@jayeshprajapaticrest 
In the log attached, EventSeverity is null even when ThreatOriginalConfidence is malicious. Can you please check if its inline with product logs.

Reply:
@vakohl 
We double checked the sample data and there were values for "ruleInfo_severity_s" in the .csv and "ruleInfo_severity" in the .json raw log using which "EventSeverity" ASIM field is filled with the Severity data.
<img width="959" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/98145046/b02c5937-68fd-46b8-b44b-06161e719fc5">

<img width="941" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/98145046/73a08b87-369f-4abb-b041-8006d8ce9cd5">

Can you please confirm again if you are available to get the value for "EventSeverity" instead of null.

Reply:
Maybe an issue with my ingested logs, all EventSeverity are coming as null. thanks for the screenshot, I think we are good.